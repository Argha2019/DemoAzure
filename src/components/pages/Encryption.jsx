import CryptoJS from 'crypto-js';

/**
 * @author utssaha(Utshab Saha)
 * 
 * This file is responsible or our own encryption and decryption. We can implement any number of 
 * algorithm and use it in other classes
 */
const secretKey = 'theNightWalkers';
const salt = "3FF2EC019C627B945225DEBAD71A01B6985FE84C95A70EB132882F88C0A59A55";
const iv = "12345678901234567890123456789012";

/**
 * This function uses AES algorithm for generating the key
 * @returns String
 */
const generateKey = () => {
    return CryptoJS.PBKDF2(secretKey, CryptoJS.enc.Hex.parse(salt), {
        keySize: 128/32,
        iterations: 10
    });
};

/**
 * This function uses AES algorithm for encryption
 * @param data 
 * @type String
 * @returns String
 */
const encryptByAES = (data) => {
    let key = generateKey();
    return CryptoJS.AES.encrypt(data, key, {
        iv: CryptoJS.enc.Hex.parse(iv)
    }).ciphertext.toString(CryptoJS.enc.Base64);
};

/**
 * This function uses AES algorithm for decryption
 * @param data
 * @type String
 * @returns String
 */
const decryptByAES = (data) => {
    let key = generateKey();
    let cipherParams = CryptoJS.lib.CipherParams.create({
        ciphertext: CryptoJS.enc.Base64.parse(data)
    });
    return CryptoJS.AES.decrypt(cipherParams, key, {
        iv: CryptoJS.enc.Hex.parse(iv)
    }).toString(CryptoJS.enc.Utf8);
};

export { encryptByAES, decryptByAES };
