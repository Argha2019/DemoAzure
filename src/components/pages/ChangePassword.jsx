import React from 'react';
import axios from 'axios';
import AgentTemplate from "../templates/AgentTemplate/Template.jsx";
import AdminTemplate from "../templates/AdminTemplate/Template.jsx";
import CustomerTemplate from "../templates/CustomerTemplate/Template.jsx";
import {Button, Form, Col, Row} from 'react-bootstrap';
import Cookies from 'js-cookie';
import '../../css/global.css';
import '../../css/registerCSS.css';
import { decryptByAES, encryptByAES } from './Encryption.jsx';
import Config from "../../config/urls.jsx";


const passwordRegex = RegExp(/^(?=.*[a-z])(?=.*[0-9])(?=.*[A-Z])(?=.*[@#$%!]).{8,40}$/);

const formValid = ({ formErrors, ...rest }) => {
	let valid = true;
	Object.values(formErrors).forEach(val => {
    	val.length > 0 && (valid = false);
  	});

  	Object.values(rest).forEach(val => {
    	val === null && (valid = false);
  	});
	return valid;
};

export default class ChangePassword extends React.Component {
	constructor(props){
    	super(props)
    	this.changePassword=this.changePassword.bind(this);
  		this.handleChange=this.handleChange.bind(this);

	    this.state = {
    		username: '',
			oldPassword: '',
			newPassword: '',
			confirmPassword: '',
			validCredential: '',
			changePasswordResultMsg: '',
			formErrors: {
      			oldPassword:'',
				newPassword: '',
				confirmPassword: ''
    		}
  		};
	}

	handleChange(e) {
  		e.preventDefault();
  		const {name, value} = e.target;
  		let formErrors = { ...this.state.formErrors};
  		switch(name) {
    		case "oldPassword":
    			formErrors.oldPassword =
    			(oldPassword.value === newPassword.value) ? "New Password must not match Old Password" : "";
    		break;
    		case "newPassword":
	   			formErrors.confirmPassword =
				(newPassword.value !== confirmPassword.value) ? "Entered password doesn't match with New Password" : "";
      			formErrors.newPassword =
	  			!(passwordRegex.test(value)) ? "Password must have 8 or more characters and must have at least a uppercase letter, lowercase letter, number and special character" : '';
      			formErrors.oldPassword =
      			(oldPassword.value === newPassword.value) ? "New Password must not match Old Password" : "";
      		break;
    		case "confirmPassword":
      			formErrors.confirmPassword =
      			(newPassword.value !== confirmPassword.value) ? "Entered password doesn't match with New Password" : "";
	  		break;
    		default:
      		break;
 		}
    	this.setState({validCredential: ""});
    	this.setState({ formErrors, [name]: value });
	}

	changePassword(e) {
  		e.preventDefault();
  		if(formValid(this.state)) {
			let username = '';
			if(Cookies.get('adminData')) {
				username = decryptByAES(JSON.parse(decryptByAES(Cookies.get('adminData')))[0].username)
			}
			else if(Cookies.get('agentData')) {
				username = decryptByAES(JSON.parse(decryptByAES(Cookies.get('agentData')))[0].username)
			}
		 	else {
				username = decryptByAES(JSON.parse(decryptByAES(Cookies.get('customerData')))[0].username)
			}
    	let oldPassword = encryptByAES(this.state['oldPassword']);
   		let newPassword = encryptByAES(this.state['newPassword']);
		const password = {
        	"username" : encryptByAES(username),
			"currentPassword" : oldPassword,
			"newPassword" : newPassword
    	};
		var url = Config.serverUrl + '/signup/password';
		axios.put(url, password)
     	.then(res => {
   			if(res.data===true) {
				this.setState({changePasswordResultMsg: "Password changed successfully!"});
				this.setState({validCredential: ""});
	   			if(Cookies.get('adminData')) {
   					Cookies.remove('adminData');
	   	 		}
	   			else if(Cookies.get('agentData')) {
		   			Cookies.remove('agentData');
	    		}
				else {
					Cookies.remove('customerData');
				}
				setTimeout(() => {
					this.props.history.push('/')
				  }, 1500);
			}
			else {
    			this.setState({validCredential: "Old password is not correct!"});
 			}
 		}
 		).catch(err => {
         	this.setState({validCredential: "Old password is not correct!"});
			});
  		}

   	 	else {
    		this.setState({validCredential: "Please enter valid Credentials"});
   		}
	}

	render(){
  		const { formErrors } = this.state;

  		if(Cookies.get('adminData')) {
			return (
				<AdminTemplate>
				
				<div className="register-wrap" style={{height: "600px"}}>
					<br/>
					<br/>
					<br/>
					<br/>
				<h1 className="fontWhite">Change Password</h1>
  				<div align="center" className="Oms register-html">
			    <span className="bigSuccessMessage">{this.state.changePasswordResultMsg}</span>
				<span className="bigErrorMessage">{this.state.validCredential}</span>
				<Form className="MyForm register-form" onSubmit={this.changePassword}>
				<Form.Group as={Row} controlId="oldPassword">
				<Col sm={5} md={5}>
				<Form.Label className="Labelling label">Old Password*</Form.Label>
				</Col>
				<Col  sm={7} md={7}>
				<Form.Control name="oldPassword" type="password" placeholder="Old Password" ref="username" onChange={this.handleChange} required/>
				</Col>
				</Form.Group>
				{formErrors.oldPassword.length > 0 && (
	  				<span className="errorMessage">{formErrors.oldPassword}</span>
				)}
				<Form.Group as={Row} controlId="newPassword">
				<Col sm={5} md={5}>
				<Form.Label className="Labelling label">New Password*</Form.Label>
				</Col>
				<Col sm={7} md={7}>
				<Form.Control name="newPassword" type="password" onCopy="return false" onCut="return false" placeholder="New Password" ref="newPassword" onChange={this.handleChange} required/>
				</Col>
				</Form.Group>
				{formErrors.newPassword.length > 0 && (
	  				<span className="errorMessage">{formErrors.newPassword}</span>
				)}
				<Form.Group as={Row} controlId="confirmPassword">
				<Col sm={5} md={5}>
				<Form.Label className="Labelling label">Confirm Password*</Form.Label>
				</Col>
				<Col sm={7} md={7}>
				<Form.Control name="confirmPassword" type="password" placeholder="Confirm Password" ref="confirmPassword" onChange={this.handleChange} required/>
				</Col>
				</Form.Group>
				{formErrors.confirmPassword.length > 0 && (
	  				<span className="errorMessage">{formErrors.confirmPassword}</span>
				)}
				<br></br>
				<br></br>
				<p className="mandatoryFields">* Fields are mandatory</p>
				<Button className="signIn" variant="primary" type="submit">Save Changes</Button>
				</Form>
				</div>
				</div>
				  </AdminTemplate>
			);
		}

 		else if(Cookies.get('agentData')){
    		return(
        		<AgentTemplate>
				<div className="register-wrap" style={{height: "650px"}}>
				<br/>
					<br/>
					<br/>
					<br/>
          		<h1 className="fontWhite">Change Password</h1>
				 
        		<div align="center" className="Oms register-html">
				<span className="bigSuccessMessage">{this.state.changePasswordResultMsg}</span>
				<span className="bigErrorMessage">{this.state.validCredential}</span>
				<Form className="MyForm register-form" onSubmit={this.changePassword}>
          		<Form.Group as={Row} controlId="oldPassword">
         		<Col sm={5} md={5}>
          		<Form.Label className="Labelling label">Old Password*</Form.Label>
          		</Col>
          		<Col  sm={7} md={7}>
          		<Form.Control name="oldPassword" type="password" placeholder="Old Password" ref="username" onChange={this.handleChange} required/>
          		</Col>
          		</Form.Group>
          		{formErrors.oldPassword.length > 0 && (
            		<span className="errorMessage">{formErrors.oldPassword}</span>
		  		)}
          		<Form.Group as={Row} controlId="newPassword">
          		<Col sm={5} md={5}>
          		<Form.Label className="Labelling label">New Password*</Form.Label>
          		</Col>
          		<Col sm={7} md={7}>
          		<Form.Control name="newPassword" type="password" onCopy="return false" onCut="return false" placeholder="New Password" ref="newPassword" onChange={this.handleChange} required/>
          		</Col>
          		</Form.Group>
          		{formErrors.newPassword.length > 0 && (
            		<span className="errorMessage">{formErrors.newPassword}</span>
		  		)}
		  		<Form.Group as={Row} controlId="confirmPassword">
          		<Col sm={5} md={5}>
          		<Form.Label className="Labelling label">Confirm Password*</Form.Label>
          		</Col>
          		<Col sm={7} md={7}>
          		<Form.Control name="confirmPassword" type="password" placeholder="Confirm Password" ref="confirmPassword" onChange={this.handleChange} required/>
          		</Col>
          		</Form.Group>
          		{formErrors.confirmPassword.length > 0 && (
           			<span className="errorMessage">{formErrors.confirmPassword}</span>
          		)}
          		<br></br>
		  		<br></br>
		  		<p className="mandatoryFields">* Fields are mandatory</p>
          		<Button className="signIn" variant="primary" type="submit">Save Changes</Button>
      			</Form>
      			</div>
				  </div>
        		</AgentTemplate>
			);
		}

		else {
			return(
				<CustomerTemplate>
				<div className="register-wrap" style={{height: "600px"}}>
				<br/>
					<br/>
					<br/>
					<br/>
	  			<h1 className="fontWhite">Change Password</h1>
				 
				<div align="center" className="Oms register-html">
				<span className="bigSuccessMessage">{this.state.changePasswordResultMsg}</span>
				<span className="bigErrorMessage">{this.state.validCredential}</span>
				<Form className="MyForm register-form" onSubmit={this.changePassword}>
	  			<Form.Group as={Row} controlId="oldPassword">
	  			<Col sm={5} md={5}>
	  			<Form.Label className="Labelling label">Old Password*</Form.Label>
	  			</Col>
	  			<Col  sm={7} md={7}>
	  			<Form.Control name="oldPassword" type="password" placeholder="Old Password" ref="username" onChange={this.handleChange} required/>
	  			</Col>
	  			</Form.Group>
	  			{formErrors.oldPassword.length > 0 && (
					<span className="errorMessage">{formErrors.oldPassword}</span>
	  			)}
	  			<Form.Group as={Row} controlId="newPassword">
	  			<Col sm={5} md={5}>
	  			<Form.Label className="Labelling label">New Password*</Form.Label>
	 			</Col>
	  			<Col sm={7} md={7}>
	  			<Form.Control name="newPassword" type="password" onCopy="return false" onCut="return false" placeholder="New Password" ref="newPassword" onChange={this.handleChange} required/>
	  			</Col>
	  			</Form.Group>
	  			{formErrors.newPassword.length > 0 && (
					<span className="errorMessage">{formErrors.newPassword}</span>
	  			)}
	  			<Form.Group as={Row} controlId="confirmPassword">
	  			<Col sm={5} md={5}>
	  			<Form.Label className="Labelling label">Confirm Password*</Form.Label>
	  			</Col>
	  			<Col sm={7} md={7}>
	  			<Form.Control name="confirmPassword" type="password" placeholder="Confirm Password" ref="confirmPassword" onChange={this.handleChange} required/>
	  			</Col>
	  			</Form.Group>
	  			{formErrors.confirmPassword.length > 0 && (
					<span className="errorMessage">{formErrors.confirmPassword}</span>
	  			)}
	  			<br></br>
	  			<br></br>
	  			<p className="mandatoryFields">* Fields are mandatory</p>
	  			<Button className="signIn" variant="primary" type="submit">Save Changes</Button>
  				</Form>
  				</div>
				  </div>
				</CustomerTemplate>
			);
		}
	}
}



