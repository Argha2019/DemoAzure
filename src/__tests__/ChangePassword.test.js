import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Cookies from 'js-cookie';
import { encryptByAES , decryptByAES } from '../components/pages/Encryption.jsx';
import ChangePassword from '../components/pages/ChangePassword.jsx';

describe('ChangePassword Component', () => {
   it('Should render snapshot for Customer', () => {
      const wrapper = shallow(<ChangePassword />);
      expect(toJson(wrapper)).toMatchSnapshot();
	});
	it('Should render snapshot for Admin', () => {
		Cookies.set('adminData',"amit");
		const wrapper = shallow(<ChangePassword />);
		expect(toJson(wrapper)).toMatchSnapshot();
		Cookies.remove('adminData',"amit");
	  });
	  it('Should render snapshot Agent', () => {
		Cookies.set('agentData',"amit");
		const wrapper = shallow(<ChangePassword />);
		expect(toJson(wrapper)).toMatchSnapshot();
		Cookies.remove('agentData','amit');
	  });

it('Valid Credential Check', () => {
	const wrapper = shallow(<ChangePassword />);
	const form = wrapper.find('.MyForm');
	const event = {
	  preventDefault: () => {}
	};

	 wrapper.setState({

	   formErrors: {oldPassword:"",newPassword:"Entered password doesn't match with New Password",confirmPassword:""}
	 });
	 wrapper.update();

	const formSubmit = form.find('.MyForm');
	formSubmit.simulate('submit', event);
	  expect(wrapper.state('validCredential')).toEqual("Please enter valid Credentials");
	});


});




