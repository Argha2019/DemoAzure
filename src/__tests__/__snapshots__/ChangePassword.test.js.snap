// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword Component Should render snapshot Agent 1`] = `
<AgentTemplate>
  <div
    className="register-wrap"
    style={
      Object {
        "height": "650px",
      }
    }
  >
    <br />
    <br />
    <br />
    <br />
    <h1
      className="fontWhite"
    >
      Change Password
    </h1>
    <div
      align="center"
      className="Oms register-html"
    >
      <span
        className="bigSuccessMessage"
      />
      <span
        className="bigErrorMessage"
      />
      <Form
        className="MyForm register-form"
        inline={false}
        onSubmit={[Function]}
      >
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="oldPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Old Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="oldPassword"
              onChange={[Function]}
              placeholder="Old Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="newPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              New Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="newPassword"
              onChange={[Function]}
              onCopy="return false"
              onCut="return false"
              placeholder="New Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="confirmPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Confirm Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Confirm Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <br />
        <br />
        <p
          className="mandatoryFields"
        >
          * Fields are mandatory
        </p>
        <Button
          active={false}
          className="signIn"
          disabled={false}
          type="submit"
          variant="primary"
        >
          Save Changes
        </Button>
      </Form>
    </div>
  </div>
</AgentTemplate>
`;

exports[`ChangePassword Component Should render snapshot for Admin 1`] = `
<AdminTemplate>
  <div
    className="register-wrap"
    style={
      Object {
        "height": "600px",
      }
    }
  >
    <br />
    <br />
    <br />
    <br />
    <h1
      className="fontWhite"
    >
      Change Password
    </h1>
    <div
      align="center"
      className="Oms register-html"
    >
      <span
        className="bigSuccessMessage"
      />
      <span
        className="bigErrorMessage"
      />
      <Form
        className="MyForm register-form"
        inline={false}
        onSubmit={[Function]}
      >
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="oldPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Old Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="oldPassword"
              onChange={[Function]}
              placeholder="Old Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="newPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              New Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="newPassword"
              onChange={[Function]}
              onCopy="return false"
              onCut="return false"
              placeholder="New Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="confirmPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Confirm Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Confirm Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <br />
        <br />
        <p
          className="mandatoryFields"
        >
          * Fields are mandatory
        </p>
        <Button
          active={false}
          className="signIn"
          disabled={false}
          type="submit"
          variant="primary"
        >
          Save Changes
        </Button>
      </Form>
    </div>
  </div>
</AdminTemplate>
`;

exports[`ChangePassword Component Should render snapshot for Customer 1`] = `
<CustomerTemplate>
  <div
    className="register-wrap"
    style={
      Object {
        "height": "600px",
      }
    }
  >
    <br />
    <br />
    <br />
    <br />
    <h1
      className="fontWhite"
    >
      Change Password
    </h1>
    <div
      align="center"
      className="Oms register-html"
    >
      <span
        className="bigSuccessMessage"
      />
      <span
        className="bigErrorMessage"
      />
      <Form
        className="MyForm register-form"
        inline={false}
        onSubmit={[Function]}
      >
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="oldPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Old Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="oldPassword"
              onChange={[Function]}
              placeholder="Old Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="newPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              New Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="newPassword"
              onChange={[Function]}
              onCopy="return false"
              onCut="return false"
              placeholder="New Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <FormGroup
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "noGutters": false,
              },
              "render": [Function],
            }
          }
          controlId="confirmPassword"
        >
          <Col
            md={5}
            sm={5}
          >
            <FormLabel
              className="Labelling label"
              column={false}
              srOnly={false}
            >
              Confirm Password*
            </FormLabel>
          </Col>
          <Col
            md={7}
            sm={7}
          >
            <FormControl
              name="confirmPassword"
              onChange={[Function]}
              placeholder="Confirm Password"
              required={true}
              type="password"
            />
          </Col>
        </FormGroup>
        <br />
        <br />
        <p
          className="mandatoryFields"
        >
          * Fields are mandatory
        </p>
        <Button
          active={false}
          className="signIn"
          disabled={false}
          type="submit"
          variant="primary"
        >
          Save Changes
        </Button>
      </Form>
    </div>
  </div>
</CustomerTemplate>
`;
